### PR TITLE
add shims for WithOptConstParam query calls

### PR DESCRIPTION
**Pull request by lcnr**
r? @ecstatic-morse @eddyb

### CODE REVIEW

File: compiler/rustc_middle/src/mir/query.rs [hunk 1]
```diff
@@ -440,4 +448,28 @@ impl<'tcx> TyCtxt<'tcx> {
             self.promoted_mir(def.did)
         }
     }
+
+    #[inline]
+    pub fn optimized_mir_of_opt_const_arg(
```
Comment (ecstatic-morse) [inline compiler/rustc_middle/src/mir/query.rs:453]: Some of these are suffixed with `of_opt_const_arg` while some have `opt_const_arg`. Does this denote different semantics? If not, can we use the same suffix for everything?

**Review by ecstatic-morse — COMMENTED** [review]
r=me once my question about suffixes is resolved

In my perfect world, we would encode the `unknown` variant of `WithOptConstArg` as a type-state, so you would go from `UnknownConstArg(DefId)` to `KnownConstArg(DefId, Option<DefId>)` at some point (maybe with a name change as well). I don't know how ergonomic this would be, however. Adding an `Unknown` variant to `WithOptConstArg` might be a good middle ground as eddyb mentioned on Zulip.

File: compiler/rustc_middle/src/mir/query.rs (hunk 1 above)
Comment (lcnr) [reply compiler/rustc_middle/src/mir/query.rs]: removes `of` for all of them

**Comment by lcnr** [comment]
I guess

@bors r=ecstatic-morse

**Pull request merged**